cluster-autoscaler/cloudprovider/azure/azure_scale_set.go

```diff
@@ -244,7 +243,6 @@ func (scaleSet *ScaleSet) waitForDeleteInstances(future *azure.Future, requiredI
 	isSuccess, err := isSuccessHTTPResponse(httpResponse, err)
 	if isSuccess {
 		klog.V(3).Infof("virtualMachineScaleSetsClient.WaitForAsyncOperationResult - DeleteInstances(%v) success", requiredIds.InstanceIds)
-		scaleSet.invalidateInstanceCache()
```

Comment:
success would just leave a node in not ready state, so it will either get cleaned up or next cache refresh will remove it. There's no need to be proactive about it.